### ISSUE TITLE
Expression prefix parsed instead of identifier

### ISSUE THREAD

**Issue by yoavcloud** [apache/datafusion-sqlparser-rs#1496]
When a statement uses identifiers that match certain expression prefix keywords, the parser will try to parse the expression over parsing the token as an identifier. The following statements work on Snowflake but the parser errors on the SELECT. If a different identifier name is selected, for example, `interval1` then the parser works as expected.

```sql
create or replace temporary table test(interval int);
insert into test (interval) values (1);
insert into test (interval) values (2);
select max(interval) from test;
```

**Comment by yoavcloud** [comment]
PR to fix this: https://github.com/apache/datafusion-sqlparser-rs/pull/1513

**Comment by yoavcloud** [comment]
Fixed in #1513
